core/src/main/scala/com/avast/clients/rabbitmq/DefaultRabbitMQStreamingConsumer.scala

```diff
@@ -0,0 +1,276 @@
+
+                  try {
+                    oldConsumer.channel.close()
+                  } catch {
+                    case NonFatal(e) => logger.debug(s"Could not close channel", e)
+                  }
+
+                  consumerTag
+                }
+                .flatTap(_ => tryCancelRunningTasks)
+
+            case None =>
+              F.delay {
+                logger.debug("No old consumer to be cancelled!")
+                initialConsumerTag
+              }
+          }
+          .flatMap { consumerTag =>
+            blocker.delay {
+              logger.debug("Starting consuming")
+              DefaultRabbitMQClientFactory.startConsumingQueue(newChannel, queueName, consumerTag, newConsumer)
+              ()
+            }
+          } >> isOk.set(true)
+      }
+    }
+  }
+  private lazy val close: F[Unit] = recoveringMutex.withPermit {
+    isOk.get.flatMap { isOk =>
+      if (isOk) consumer.get.flatMap {
+        case Some(streamingConsumer) =>
+          blocker.delay {
+            streamingConsumer.stopConsuming()
+            streamingConsumer.channel.close()
+          }
+
+        case None => F.unit
+      } else F.unit
+    } >> isOk.set(false) >> isClosed.set(true)
+  }
+  private lazy val tryCancelRunningTasks: F[Unit] = {
+    tasks
+      .update { tasks =>
+        tasks.foreach(_.unsafeRunSync())
+        Set.empty
+      }
+      .to[F]
+  }
+  private lazy val stopConsuming: F[Unit] = {
+    recoveringMutex.withPermit {
+      isOk.set(false) >> consumer.get.flatMap(_.fold(F.unit)(_.stopConsuming())) // stop consumer, if there is some
+    }
+  }
+  private lazy val checkNotClosed: F[Unit] = {
+    isClosed.get.flatMap(cl => if (cl) F.raiseError[Unit](new IllegalStateException("This consumer is already closed")) else F.unit)
+  }
+
+  private def handleStreamFinalize(e: ExitCase[Throwable]): F[Unit] = e match {
+    case ExitCase.Completed =>
+      stopConsuming
+        .flatTap(_ => F.delay(logger.debug(s"[$name] Delivery stream was completed")))
+
+    case ExitCase.Canceled =>
+      stopConsuming
+        .flatTap(_ => F.delay(logger.debug(s"[$name] Delivery stream was cancelled")))
+
+    case ExitCase.Error(e: ShutdownSignalException) =>
+      stopConsuming
+        .flatTap { _ =>
+          F.delay {
+            streamFailureMeter.mark()
+            logger.error(s"[$name] Delivery stream was terminated because of channel shutdown. It might be a BUG int the client", e)
+          }
+        }
+
+    case ExitCase.Error(e) =>
+      stopConsuming
+        .flatTap(_ =>
+          F.delay {
+            streamFailureMeter.mark()
+            logger.debug(s"[$name] Delivery stream was terminated", e)
+        })
+  }
+
+  private def deliveryCallback(delivery: Delivery[Bytes]): F[DeliveryResult] = {
+    for {
+      deferred <- Deferred[F, DeliveryResult]
+      consumerOpt <- this.consumer.get
```

Comment:
I like to call these `maybeConsumer`, it's quite nice ;)

Reply:
IDEA agrees with you however I don't :stuck_out_tongue: